[TASK] !!! Make extension TYPO3 12 compatible

PKuhlmay opened the pull request.

Commit:
[BUGFIX] 400 - incompatible declaration of ExtensionValidator::isValid

Commit:
[TASK] #404 - Align with new API for backend modules

Using new API for registering backend modules
Register Icon for backend module via new Icons.php

Commit:
Convert inject methods to constructor

Commit:
Rework Button bars

# Conflicts:
#	Configuration/Backend/Modules.php

Commit:
[GUI] Rework of the whole JavaScript GUI

Commit:
[TASK] #401 - Raise new versions for new TYPO3 12 release

Commit:
[PHP] #403 - Fix PHP Warning - trying to access array offset

Commit:
[BUGFIX] #402 - Fix htmlspecialchars - null given instead of array

Commit:
[TASK] 416 - Introduce rector

Commit:
Run Rector up to PHP 8.1 and TYPO3 12

Commit:
Removed undefined call of method

Commit:
settings for devbox and gitignore

Commit:
[WEBPACK] include webpack

Commit:
[GUI] Includes Source Code from former React App for EB and included it into webpack

Commit:
Update packages

oliverkroener committed:
[GUI] Fix typo for importing an scss file (#659)

Commit:
[WEBPACK] adjusting webpack config

Commit:
[GUI] Rework of the whole React JS GUI

Commit:
661 - bugfix - set properties for controller

Commit:
664 - bugfix: check for undefined array key in controller

Commit:
[RELEASE] First alpha release

Reworked the whole JS GUI
Removed deprecated code
Updated this extension to TYPO3 12 and PHP > 8.1

Commit:
669 bugfix: fix availability of EB while CSP is active

Commit:
671 - fix: availability to close modals with the close-button

Commit:
666 - bugfix: store more than only one plugin/module/author

Commit:
reformatting

Commit:
remove obsolete generation of routes inside module configuration

Commit:
670 - increase the font size

Commit:
533 - Display output dir and backup dir inside the footer of EB

Commit:
Release 12.0.0-alpha.4

Commit:
678 - make footer visible and hide button bar

Commit:
680 - storing authors inside composer.json

Commit:
663 - WIP - remove deprecated functions

Commit:
663 - remove deprecated functions

Commit:
533 - add output dir to footer

hgalt committed:
added missing type declaration

Commit:
413 - make selectbox configurable

add an input field for the selectbox, so you can predefine the selectable values

Commit:
682: fix: generated module now shows the correct labels

Commit:
WIP - fixes generation of modules

Commit:
688 - fix: make deletion of author/plugin/module possible again

Commit:
686: task: make properties inside model movable und deletable

Commit:
689: fix: scss and js generation

Commit:
693 - added a "new" button

Commit:
691 - WIP remove node button

Commit:
Toggle advanced options and GUI improvements

Commit:
413 - Separate select values by new line

Commit:
413 Map options for selectbox into own object in JSON

Commit:
695 - Make String-property configurable

Commit:
695 - fix config for type string

Commit:
696 - make type text configurable

merge type richtext into text

Commit:
702-703: make float and integer configurable

Commit:
699: make colorpicker configurable

Commit:
700 - make type password configurable

Commit:
701 - make email configurable

Commit:
704 - make boolean configurable

Commit:
711 - fix tca output for inputlink

Commit:
705 - make datetime configurable

Commit:
709 - make file configurable

Commit:
716 - make none configurable

Commit:
Bugfixes

Commit:
413 - whereClause for select type

Commit:
696 - rows for type text

Commit:
717 - make relation removable from model

Commit:
GUI - Set modal background

Commit:
Set default branch for new extension to main instead of master

Commit:
722 - Adds validation to React JS Application

Commit:
Fix PHP Error

Commit:
Fix Sass linter

Commit:
722 - disable validation for some fields

Commit:
626 - WIP - create plugins as own CType

Commit:
Removes pregenerated CSS Styles

Commit:
510 - adds icon for plugin

Commit:
remove comment in extLocalconf

Commit:
662 - fix issue with default value within NodeFactory

Commit:
adds returnInterface for actions

Commit:
Set property of model to lowercase

Commit:
fix: html templates for Models

Commit:
672 - controller names should start with uppercase

Commit:
662 - fix issue with default value for nodefactory

Commit:
683 - add moduleTemplateFactory to Controller Templates

Commit:
Styling for Backend Templates

Commit:
WIP 727 - remove test generation

Commit:
Code cleanup - removed slashes from settings.yamlt

Commit:
Code Cleanup

Commit:
626 - own CType for Plugin

Commit:
726 - bugfix: select box for relation was not set properly

Commit:
Release v12.0.0-beta.1

Commit:
Code cleanup

Commit:
Refactoring and code cleanup

Commit:
WIP Rename settingspanelcomponent

Commit:
[DOCS] Cleanup documentation

Commit:
[TASK] Render correct access value for backend modules

Commit:
GUI improvements

Commit:
727 remove test generation

Commit:
[Release] v12.0.0-beta.2

Commit:
update npm packages

Commit:
Switch documentation rendering to PHP

Commit:
[TASK] 738 - add notification when storage path is missing

Commit:
[DOCS] Fix issues in documentation

Commit:
fix editorconfig

Commit:
update packages

Commit:
Merge branch 'master' into typo3-12-reactjs

# Conflicts:
#	.editorconfig
#	.github/workflows/test-documentation.yml
#	Classes/Parser/NodeFactory.php
#	Documentation/ChangeLog/Index.rst
#	Documentation/guides.xml
#	Resources/Public/jsDomainModeling/extbaseModeling.css
#	Resources/Public/jsDomainModeling/extbaseModeling.scss
#	composer.json
#	ext_emconf.php
#	package-lock.json

Commit:
fix package-lock.json and yarn.lock

DavidBruchmann committed:
[TASK] fix several minor issues in TCA and flexfom. (#747)

* fix #746
* fix #745
* fix #744
* fix #743

Commit:
Test Coverage

Commit:
update php version inside github workflow

Commit:
set Bootstrap.php for testing

Commit:
Removes unused code

Commit:
Merge branch 'master' into typo3-12-reactjs

# Conflicts:
#	.github/workflows/tests.yml
#	composer.json

Commit:
Remove files which should never have been published to github

DavidBruchmann committed:
[TASK] sort list of local extensions for optional editing

DavidBruchmann committed:
move SCSS from Resources/Public into Build

DavidBruchmann committed:
fix #768, show another message when no extensions to open are available

Commit:
WIP refactor extension loading

Probably this should be inside branch v12-load-old-extensions

PKuhlmay committed:
WIP load old extensions

PKuhlmay committed:
WIP load old extensions

DavidBruchmann commented on composer.json:
```diff
@@ -21,20 +21,21 @@
-		"typo3/cms-backend": "^11.4",
-		"typo3/cms-core": "^11.4"
+		"nikic/php-parser": "^4.15.4",
+		"typo3/cms-backend": "^12.2.0 || 12.3.x-dev || dev-main",
```
this looks outdated

PKuhlmay replied:
Strange. This must be very old and not have been updated for some months 😅

DavidBruchmann replied:
there are newer commits with that file ...

DavidBruchmann replied:
I'd propose just merge it in a new branch and then we can check easier.
A large stack like this is hard to oversee.

PKuhlmay replied:
You dont need to read every commit. This PR is for the whole branch. There is already a seperate branch for this. 
Just look at the changed files as these lines are already updated. It was a little bit confusing.

DavidBruchmann replied:
Go for it, we need to put the things together we have till now before proceeding.

DavidBruchmann committed:
[TASK] Replace spyc yaml parser (#772)

replace SpycYAMLParser with Symfony Yaml parser, fix #755

PKuhlmay committed:
[DOCS] Adds changelog infos for v12

PKuhlmay committed:
[DOCS] Adds changelog infos for v12

PKuhlmay committed:
fix for old names from v11 or earlier

PKuhlmay committed:
fix for old names from v11 or earlier

ErHaWeb committed:
[BUGFIX] Uniform naming of property fields

This patch ensures that all fields that can occur within properties are written in the same way. In this way, values are retained when saving and reloading, for example.

ErHaWeb committed:
[BUGFIX] Revision of extension identifiers

PKuhlmay committed:
Remove comment inside settings.yamlt for staticDateInXliffFiles

DavidBruchmann committed:
Set generating empty git repository to a default 'false'

georgringer committed:
[BUGFIX] Use correct casts

() must be used to avoid still notices

PKuhlmay committed:
Current state

PKuhlmay committed:
[TESTS] Fix wrong return definition

PKuhlmay committed:
Reformat settings file for testing

PKuhlmay committed:
[BUGFIX] Fix undefined array keys

PKuhlmay committed:
WIP fix propertyname

PKuhlmay committed:
Fix Typo in author role

PKuhlmay committed:
fix loading of relations

PKuhlmay committed:
Merge branch 'eric-v12-load-old-extensions' into v12-load-old-extensions

# Conflicts:
#	Build/Sources/components/ReactFlow/CustomModelNode.jsx
#	Build/Sources/helper/converter/convertModulesToNodes.js
#	Documentation/ChangeLog/12-0-0.rst
